crates/holochain_sqlite/src/db.rs

```diff
@@ -201,12 +206,23 @@ impl<Kind: DbKindT> DbRead<Kind> {
         r
     }
 
-    async fn acquire_reader_permit(&self) -> OwnedSemaphorePermit {
-        self.read_semaphore
-            .clone()
-            .acquire_owned()
-            .await
-            .expect("We don't ever close these semaphores")
+    async fn acquire_reader_permit<E>(&self) -> Result<OwnedSemaphorePermit, E>
+    where
+        E: From<DatabaseError> + Send + 'static,
+    {
+        match tokio::time::timeout(
+            std::time::Duration::from_secs(2),
```

Comment:
I'm torn on this value. 2 seconds might be a real read time on a slow system with a cold cache and a slow hdd. Maybe 10s?

Reply:
The timeout needs to be aggressive to help the system recover when things go wrong but as we don't have any metrics currently telling us what real query times look like (once warm with a fair bit of data stored It's under 0.2s but not sure in general) it's probably safer to go higher as you suggest. At least we'll know that something is holding database handles when this times out even if Holochain doesn't recover fully. That's a good step for now